pkg/registries/registries.go

```diff
@@ -131,20 +164,35 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		}
 	}
 
-	// insecureRegistries and blockedRegistries are lists of registries; now that mirrors can be configured at a namespace/repo level,
+	// Add the blocked registry entries to the registries list so that we can find sub-scopes of insecure registries and set both the
+	// blocked and insecure flags accordingly.
+	// e.g *.blocked.insecure.com is a sub-scope of *.insecure.com and should have both the insecure and blocked options set to true. If
+	// we don't add the blocked registries list to the registries config list before going through the insecure registries we won't be able
+	// to check if *.blocked.insecure.com is a sub-scope of *.insecure.com as it won't exist in the registries config list and will not have
+	// insecure=true, so we need to populate the registries config list with the blocked registries before moving on.
+	for _, reg := range blockedRegistries {
+		getRegistryEntry(reg)
```

Comment:
Absolutely non-blocking: Maybe
```suggestion
		_ = getRegistryEntry(reg)
```
to even more motivate users to read the comment above (which does explain what is going on).

Reply:
fixed